### ISSUE TITLE
Seven-segment IPv6 addresses not handled properly

### ISSUE THREAD

**Issue by dmurvihill** [lemonlabsuk/scala-uri#195]
**Describe the bug**
The IPv6 parser seems to have misread the spec; the IPv6 string `[E873:4eC5:eBc9:9e97:6BcE:998C:95AD::]` is a valid `IPv6address` object under RFC 3986, matching the last ABNF rule in the list on page 20:

```
IPv6address =
    <other options>
    / [ *6( h16 ":" ) h16 ] "::"

h16 = 1*4HEXDIG
```

But parsing this causes an error that says there must be exactly eight or fewer than six hex pieces, which I think must be a misunderstanding since this section of the document is very confusing.

**To Reproduce**
```scala
val url = "https://[E873:4eC5:eBc9:9e97:6BcE:998C:95AD::]/"
Url.parse(url)
```

```
[info]     Message: IPv6 has too many pieces. Must be either exactly eight hex pieces or fewer than six hex pieces with a '::'    
```

**Comment by theon** [comment]
Thanks for raising this.

Yes, I think this my misinterpretation of the spec. It makes sense that the spec allows `::` still to be used as shorthand for a single `:0000` segment.

**Comment by theon** [comment]
This has been released in version `2.2.6`
